src/term.rs

```diff
@@ -439,7 +468,8 @@ fn show_precedence_cla(term: &Term, context_precedence: usize, depth: u32) -> St
                     .expect("error while printing term")
                     .to_string()
             } else {
-                from_u32(96 + *i as u32)
+                // use a different name than bound variables
+                from_u32(max_depth + 96 + *i as u32 - depth)
```

Comment:
Refer to the free variable list while considering `depth`. Also, avoid using names that might be used for bound variables by using `max_depth`.